File: 007/Problem7.c
#include <stdio.h>
#include <time.h>
#include <stdbool.h>

bool isPrime(int num) {
    for (int i = 2; i < num; i++) {
        if (num % i == 0) {
            return false;
        }
    }
    return true;
}

void problem() {
    int count = 2;
    int i = 4;
    for (;;) {
        i += 1;
        if (isPrime(i)) {
            count += 1;
            if (count == 10001) {
                break;
            }
        }
    }
    printf("%d\n", i);
}

int main() {
    clock_t t;
    t = clock();
    for (int i = 0; i < 100; i++) {
        problem();
    }
    t = clock() - t;
    double time_taken = ((double) t ) / CLOCKS_PER_SEC;
    printf("Function took %f \n", time_taken);
    return 0;
}

File: 007/Problem7.kt
import kotlin.system.measureNanoTime

fun main() {
    val loops = 10
    val timeInNanos = measureNanoTime {
        for (i in 1..loops) {
            problem()
        }
    }
    println("Function Took $timeInNanos Nano seconds for $loops loops")
}

fun isPrime(num: Int): Boolean {
    for (i in 2 until num) {
        if (num % i == 0) {
            return false
        }
    }
    return true
}

fun problem() {
    var count = 2
    var i = 4
    while (true) {
        i++
        if (isPrime(i)) {
            count++
            if (count == 10_001) {
                break
            }
        }
    }
    println(i)
}